### DISCUSSION TITLE
BFUS 1080p - chinese version (Problem solved!)

### DISCUSSION THREAD

**Discussion by uklatt** [roleoroleo/yi-hack-Allwinner-v2#383] [Q&A] [answered]
Hello,

I have a chinese version of Yi 1080p home camera (type BFUS= y21ga).
Firmware is 9.0.19.02_202007141512
This is not pairable with international app. (the chinese version doesn't work either)

I can install yi-hack, but because of the pairing with the yi app doesnt work, the connection is missing after a reboot.

Can I install an international version on this camera?
How can I install yi-hack on this camera?

----

It looks like the camera needs to complete the pairing process before working normal.
This seems to be done by **dispatch** process. This creates the link /tmp/audio for chinese or us audio files.
How can I simulate the pairing?
Would it be a good idea, to use a mtdblock3.bin (home) from an international cam?

----

Finally I used a copy of **mtdblock6** and **mtdblock7** from another working international camera.
Now the camera "speaks" english and can be used with yi-hack without problems ;-)

Uwe

**Comment by roleoroleo** [comment, marked as answer]
Maybe, change the serial number from CN to US is enough?

> Finally I used a copy of **mtdblock6** and **mtdblock7** from another working international camera.
> Now the camera "speaks" english and can be used with yi-hack without problems ;-)

Now, you should have 2 cams with the same serial.
Check if the mac address is different!

**Reply by uklatt** [reply]
Yes, I forgot to change the serial number before flashing, but I don't use the cloud. 8-)
At first I just changed mtdblock6, but that's not enough.
The MAC is still different! 

Uwe

**Comment by okilo92** [comment]
Hello,

i have a similar situation. My camera started speaking in Chinese and I can't connect it to the app and install the hack. I got the  mtdblock6 and mtdblock7 files, how can I use them to make it speak English? Model is BFUS.

**Comment by okilo92** [comment]
The camera has been running smoothly since January. Days ago I came home and found the flashing yellow LED and I hear the camera speak in Chinese (I think it says "Wait to connect"). Trying to connect it to the YI Home app, it connects to the Wifi and the led turns blue. After a few minutes the app and the camera tell me that an error has occurred. I contacted support and they sent me a bin file but it didn't work. I have tried multiple sd cards. I would like to try to restore the English language so I can understand what it says.
________________________________

**Reply by roleoroleo** [reply]
Could you send me this bin file?

**Reply by okilo92** [reply]
Yes, this Is the file they sent me: https://help.yitechnology.com/attachments/token/U6V0VxK2LNbdHLRmHXQtfRgxG/?name=one_y21ga.bin

**Reply by okilo92** [reply]
By contacting support directly on the purchase site, they asked me for the product DID and sent me another file. I don't know if it's the same, the name is the same. I compressed it so I could send it to you.
[one_y21ga.bin.zip](https://github.com/roleoroleo/yi-hack-Allwinner-v2/files/9482813/one_y21ga.bin.zip)

**Comment by cjgarciao** [comment]
Hello,
I've a similar problem. 
I bought an Outdoor 1080p camera and it spoke Chinese. When I contacted support they told me they would send me another one, and to take the one that spoke Chinese for recycling. Now I have two cameras. The one that speaks in Chinese works perfectly with the hack, only the serial number does not appear.
I had thought of overwriting partitions 6 and 7 with the data from the one that speaks English, but putting the correct serial number (I have it in the box) and the MAC if necessary. Will it work or will I have to change anything else? Are partitions 6 and 7 raw data or is it a file system?
Regards,
Carlos.

**Reply by roleoroleo** [reply]
Partition 6 and 7 are row partitions, no file system (afaik).
Probably you don't need to change anything else.
Is the serial number like xxCHxxxxxxxx?

**Reply by cjgarciao** [reply]
No, the original chinese firmware has nothing in the serial number area, only 'FF'.
Thanks.
Carlos.

**Reply by roleoroleo** [reply]
Never seen, probably an error during production.

**Reply by cjgarciao** [reply]
I have updated 'mtdblock6' and 'mtdblock7' with the version with the changed serial number. I have also restored the 'home' and 'backup' partitions (not sure if removing the SD card is enough to remove the hack). The camera can't be registered in the Yi Home app, although it now speaks English.
With the hack it works fine and now the serial number is displayed correctly.
Thanks @roleoroleo, the hack is great!
Carlos.

**Reply by MaryCam1** [reply]
> The camera can't be registered in the Yi Home app, although it now speaks English.

I've the same problem. But I need to connect to the app too. What oder changes do it need ?

**Reply by roleoroleo** [reply]
It should be enough.
The serial is stored only in mtd6 and mtd7.

**Comment by MaryCam1** [comment]
> Finally I used a copy of **mtdblock6** and **mtdblock7** from another working international camera.

Is there a guide how to do this step by step? I am using Windows.
Unfortunately I have the same problem described above, same type of camera, same firmware... Seems to be a recurring problem.
I want to solve the problem by overwriting the mtdblock containing the serNumber with the original.
Would be really grateful if someone could help me please.

**Reply by roleoroleo** [reply]
If you are able to access the cam with a shell, a dd is enough:
```
dd if=mybackupfilemtd6.bin of=/dev/mtdblock6
dd if=mybackupfilemtd7.bin of=/dev/mtdblock7
```

**Reply by MaryCam1** [reply]
Thnx for your suggestion.
I have to figure out how to proceed, then I'll try it.

**Reply by MaryCam1** [reply]
Hi, I'm a bit confused.
Before copying the mtdblocks I wanted to explore the file system of the cam.
I was able to access the cam with PuTTY.
Although I'm not really familiar with linux, I was courious tried to find out the contents of the mtdblocks and was really surprised, I found that the serial BFUSY4484T9NE4200810 is still there, in mtdblock7, while the hack Status shows 0000000000000000000000. How can that be. Of course I'm not sure if the serial is in the right place but I assume so.
What do you suggest? Proceed with dd mtdblocks anyway? Do you think that's enough?
![Immagine](https://user-images.githubusercontent.com/121692963/211219063-69330b22-e4a6-44e9-85a2-7db74c283db8.png)

**Reply by tummerry** [reply]
how to change  xxCHxxxxxxxx   xxUSxxxxxxxx?
Is there a method and procedure?
please tell me roleoroleo

**Reply by roleoroleo** [reply]
The hack reads the serial number from mtdblock6, because the structure is simpler.
I don't know what is the "master" repository, mtdblock6 or mtdblock7.
Probably overwriting mtdblock6 is enough.

**Reply by roleoroleo** [reply]
> how to change xxCHxxxxxxxx xxUSxxxxxxxx? Is there a method and procedure? please tell me roleoroleo

Simply dump the partition with dd, edit it and restore with dd.

**Reply by tummerry** [reply]
mtdblock6 or mtdblock7 where to find

**Reply by roleoroleo** [reply]
Access to the cam with a ssh client (putty) and dump them with these commands:
```
dd if=/dev/mtdblock6 of=/tmp/sd/mtdblock6.bin
dd if=/dev/mtdblock7 of=/tmp/sd/mtdblock7.bin
```

**Reply by tummerry** [reply]
ok

**Reply by tummerry** [reply]
i did but not SN
<img width="461" alt="image" src="https://user-images.githubusercontent.com/66394776/212251051-6aea8099-d89b-4bad-bb8b-f3e8025e1b66.png">

**Reply by tummerry** [reply]
teach me please

**Reply by MaryCam1** [reply]
> If you are able to access the cam with a shell, a dd is enough:
> 
> ```
> dd if=mybackupfilemtd6.bin of=/dev/mtdblock6
> dd if=mybackupfilemtd7.bin of=/dev/mtdblock7
> ```

I've copied the mtdblocks with success. Cam now speaks english, hack works fine, but cam still don't connect to the app ("Disable Cloud" is NOT enabled). What more changes do it need to connect to the app ?

**Reply by tummerry** [reply]
> > If you are able to access the cam with a shell, a dd is enough:
> > ```
> > dd if=mybackupfilemtd6.bin of=/dev/mtdblock6
> > dd if=mybackupfilemtd7.bin of=/dev/mtdblock7
> > ```
> 
> I've copied the mtdblocks with success. Cam now speaks english, hack works fine, but cam still don't connect to the app ("Disable Cloud" is NOT enabled). What more changes do it need to connect to the app ?

Please tell me how do you do the steps.
please tell me

**Reply by MaryCam1** [reply]
- performed a backup from a good camera (wiki yi-hack-Allwinner-v2)
- copied mtdblock6.bin and mtdblock7.bin to the root dir of the sd of the cam to be changed
- fixed the serial in the both mtdblocks with my favorite editor
- inserted the sd to the cam to be changed and restarted the cam
- accessed to the cam with a telnet client (putty) and performed these commands:
dd if=/tmp/sd/mtdblock6.bin of=/dev/mtdblock6
dd if=/tmp/sd/mtdblock7.bin of=/dev/mtdblock7

thats it

**Reply by tummerry** [reply]
i can do it thank you very much

**Comment by tummerry** [comment]
how to change xxCHxxxxxxxx xxUSxxxxxxxx?
Have WIKI give me see? 
roleoroleo help me

**Reply by MaryCam1** [reply]
simply, before copy mtdblocks to the cam to be restored access to the mtdblocks on the sd from the PC and change the serial with your favorite editor

**Reply by roleoroleo** [reply]
You could use hxd (https://mh-nexus.de/en/hxd/) to edit the file and change your serial number.
About the problem of the connection to the app, I don't know how to help you.
Is the serial used by another cam. If yes, I think it can't work.

You should try to kill cloud process and run it manually from a ssh session.
You should see a verbose output.

**Reply by MaryCam1** [reply]
Hi, I will give a try to the editor.
I'm sure, the serial is not used by another cam.
But I also realized that by a new start of the cam, wifi procedure must be done again.
So, the wifi informations are not stored.
Could you plase also tell me, how to kill the cloud process and run it manually ?
Thank you in advance.

**Reply by roleoroleo** [reply]
To kill cloud application:
```
kill cloud
```
to restart it:
```
cd /home/app
./cloud
```

**Reply by MaryCam1** [reply]
tried to kill cloud but got "ash: invalid number 'cloud'", seems cloud isn't started yet, then I started ./cloud, camera tried to connect to server but it failed, I got the following output, but no idea where to look...
[report cloud.txt](https://github.com/roleoroleo/yi-hack-Allwinner-v2/files/10539877/report.cloud.txt)
